web/app/habit/checkin/components/checkinRun.tsx

```diff
@@ -195,43 +258,46 @@ export default function RunCheckIn({ challenge }: { challenge: ChallengeWithChec
 
       {/* goal description */}
       <div className="w-full justify-start p-6 py-2 text-start">
-        <div className="text-xl font-bold text-dark"> Description </div>
+        <div className="text-base font-bold text-dark"> Description </div>
         <div className="text-sm text-primary"> {challenge.description} </div>
       </div>
 
       <div className="w-full justify-start p-6 py-2 text-start">
-        <div className="text-xl font-bold text-dark"> Staked Amount </div>
+        <div className="text-base font-bold text-dark"> Staked Amount </div>
         <div className="flex text-sm text-primary">
           {`${formatUnits(challenge.stake, 6)} USDC`}{' '}
         </div>
       </div>
 
-      <div className="w-full justify-start p-6 py-2 text-start">
-        <div className="text-xl font-bold text-dark"> Invite Others </div>
-        <InviteLink accessCode={challenge.accessCode} challengeId={challenge.id} />
-      </div>
-
-      <div className="m-4 text-center font-londrina text-base">
-        ⏰ Challenge {challenge.endTimestamp > now() / 1000 ? 'settles' : 'settled'}{' '}
-        {moment.unix(challenge.endTimestamp).fromNow()}
+      {/* New countdown badge with reduced margins */}
+      <div className="my-2 flex w-full justify-center">
+        <div className="inline-block rounded-full border-2 border-dotted border-orange-400 bg-orange-100 px-3 py-1">
+          <span className="text-xs font-semibold text-orange-700">
+            ⏰ Challenge {challenge.endTimestamp > now() / 1000 ? 'ends' : 'ended'} in
+          </span>
+          <span className="ml-1 text-xs font-bold text-orange-800">
+            {getCountdownString(challenge.endTimestamp)}
+          </span>
+        </div>
       </div>
 
       {/* middle section: if timestamp is not valid, show warning message */}
 
-      {/* no address detected: ask user to connect */}
-      {!address && <ConnectButton className="w-3/4" />}
-
-      {challenge.status === UserChallengeStatus.Ongoing && verifierConnected && canCheckInNow && (
-        <div className="flex w-full justify-center px-2">
-          <ActivityDropDown
-            isDisabled={!address}
-            fields={fields}
-            onActivitySelect={setChosenActivityId}
-            loading={stravaLoading}
-            activities={activitiesToUse}
-            usedActivities={activityMap[challenge.id]}
-          />
-        </div>
+      {challenge.status === UserChallengeStatus.Ongoing && canCheckInNow && (
+        <>
+          {renderCheckInOptions()}
+          {challenge.allowSelfCheckIn && (
```

Comment:
I think this part can be separate into component with `renderCheckInOptions` in the future.